src/ng/components/template/template.path.js

```diff
@@ -0,0 +1,55 @@
+import AppError from '../../../core/infrastructure/error';
+import {isUndefined} from '../../../core/services/utility';
+import {TEMPLATE_NAME} from '../../../definition';
+
+const resolvers = {};
+export default class TemplatePath {
+	constructor() {
+	}
+
+	static register(name, resolve) {
+		if (resolvers.hasOwnProperty(name)) {
+			throw new AppError(
+				`template.path`,
+				`"${name}" is already registered`);
+		}
+
+
+		resolvers[name] = resolve;
+		return TemplatePath;
+	}
+
+	static get(source) {
+		const path = this.find(source);
```

Comment:
they unfortunate static for now, because i couldnt come up how to require it to controller differently, i think it'll be evolved